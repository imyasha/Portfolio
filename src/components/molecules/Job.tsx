import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import camelCase from 'camelcase';
import { getTextColor } from 'Templates/Layout';

interface IProps {
  children: React.ReactNode;
  title: string;
  company: {
    name: string;
    url: string;
  };
  place: string;
  from: string;
  to: string;
}

const Job = ({
  children,
  title,
  company: { name: companyName, url: companyUrl },
  place,
  from,
  to,
}: IProps) => (
  <Listing data-testid={`job${camelCase(companyName, { pascalCase: true })}`}>
    <h1>{title}</h1>
    <h2>
      <strong>
        <a href={companyUrl} target="_blank" rel="noopener noreferrer">
          {companyName}
        </a>
      </strong>{' '}
      (in {place})
    </h2>
    {children}
    <small>
      {from} to {to}
    </small>
  </Listing>
);

const Listing = styled.article`
  border-bottom: 1px solid ${getTextColor};
  padding-bottom: 2rem;
  margin-bottom: 1rem;
  break-inside: avoid;

  h1 {
    margin: 0 0 0.5rem;
  }
  h2 {
    font-weight: normal;
    font-size: 1rem;
    margin: 0 0 0.5rem;
  }
`;

Job.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  title: PropTypes.string.isRequired,
  company: PropTypes.shape({
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  }).isRequired,
  place: PropTypes.string.isRequired,
  from: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
};

export default Job;
