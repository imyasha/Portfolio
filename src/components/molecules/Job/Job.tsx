import { IJob } from '@types';

import { format } from 'date-fns';
import Image from 'next/image';
import { FC, ReactNode } from 'react';

import { Box } from 'Atoms/Box';

interface JobProps extends Omit<IJob, 'description'> {
	children?: ReactNode;
}

const Job: FC<JobProps> = ({
	company,
	jobTitle,
	fromDate,
	toDate,
	technologies,
	children,
}) => {
	return (
		<Box>
			<div className="flex gap-4 items-start md:items-center">
				{company.logo && (
					<Image
						src={company.logo}
						alt={company.name}
						width={64}
						height={64}
						className="rounded-md"
					/>
				)}
				<div className="w-full mb-2">
					<div className="flex flex-col-reverse md:flex-row justify-between">
						<h3 className="font-bold">{jobTitle}</h3>
						<span className="text-sm md:text-base">
							{`${format(new Date(fromDate), 'MMM yyyy')} â€” ${
								toDate ? format(new Date(toDate), 'MMM yyyy') : 'present'
							}`}
						</span>
					</div>
					<h4>
						{company.url ? (
							<a
								href={company.url}
								target="_blank"
								rel="noopener noreferrer"
								className="underlined relative"
							>
								{company.name}
							</a>
						) : (
							<>{company.name}</>
						)}
					</h4>
					<p className="hidden md:block text-sm">
						<strong>Technologies: </strong>
						{technologies.join(', ')}
					</p>
				</div>
			</div>
			<p className="md:hidden text-sm">
				<strong>Technologies: </strong>
				{technologies.join(', ')}
			</p>
			{children}
		</Box>
	);
};

export { Job };
