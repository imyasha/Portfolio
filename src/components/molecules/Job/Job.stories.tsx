import { IJob } from '@types';

import { Job } from './Job';
import { mockJobs } from 'MockData';

export default {
	title: 'molecules/Job',
	component: Job,
};

export const JobStory = (args: Omit<IJob, 'description'>) => <Job {...args} />;

JobStory.storyName = 'Job';
JobStory.args = {
	...mockJobs[0],
	children: <div>Some body</div>,
};
