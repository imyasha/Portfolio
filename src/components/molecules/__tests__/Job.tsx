import React from 'react';
import renderer from 'react-test-renderer';

import Job from '../Job';

const job = {
  title: 'Web Developer',
  company: {
    name: 'My Company',
    url: 'test.com',
  },
  place: 'London',
  from: 'Jan 2010',
  to: 'present',
};

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Job
        title={job.title}
        company={job.company}
        place={job.place}
        from={job.from}
        to={job.to}
      >
        List of tasks
      </Job>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
