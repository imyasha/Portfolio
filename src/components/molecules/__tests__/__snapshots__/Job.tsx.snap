// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article
  className="sc-EHOje ccXLoz"
>
  <h1>
    Web Developer
  </h1>
  <h2>
    <strong>
      <a
        href="test.com"
        target="_blank"
      >
        My Company
      </a>
    </strong>
     
    (in 
    London
    )
  </h2>
  List of tasks
  <small>
    Jan 2010
     to 
    present
  </small>
</article>
`;
